src/commands/ml-models-list.ts

```diff
@@ -1,12 +1,49 @@
+import * as clc from "cli-color";
 import { Command } from "../command";
+import * as getProjectId from "../getProjectId";
 import { FirebaseModel } from "../ml/models";
+import { requirePermissions } from "../requirePermissions";
+import { ensureFirebaseMlApiEnabled, getTableForModelList } from "../ml/mlHelper";
+import * as mlApi from "../ml/mlApi";
+import * as logger from "../logger";
+import * as utils from "../utils";
+import { logPrefix } from "../ml/mlHelper";
 
 export default new Command("ml:models:list")
-  .description("lists all the Firebase ML Models in the current project.")
+  .description("list all the Firebase ML Models in the current project.")
+  .help("TODO")
   .option("-f --filter <filterString>", "(optional) A valid filterString.")
+  .before(requirePermissions, ["firebaseml.models.list"])
+  .before(ensureFirebaseMlApiEnabled)
   .action(
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async (options: any): Promise<FirebaseModel[]> => {
-      throw new Error("Not Implemented");
+    async (options: any): Promise<{ models: FirebaseModel[] }> => {
+      const projectId = getProjectId(options);
+      const models: FirebaseModel[] = await mlApi.listModels(projectId, options);
+
+      if (models.length < 1) {
```

Comment:
nit: make this a single if:
```
if (models.length < 1 && options.filter)
```

Reply:
I'm not sure I understand. Do you mean this?

if (models.length < 1 && options.filter) {
  // log message "no models matching filter"
 return { models: [] }
} else if (models.length < 1 && ! options.filter) {
  // log message "no models in project"
 return { models: [] }
}
because that doesn't seem better to me? Am I missing something?

Reply:
Ah got it, my mistake